docs/configuration.rst

```diff
@@ -153,6 +161,23 @@ Some more examples:
     have multiple components that share configuration like credentials and
     hostnames.
 
+``config('port', parser=int, doc='The port you want this to listen on.')``
+    You can provide a ``doc`` argument which will give users users who are trying to
+    configure your software a more helpful error message when they hit a configuration
+    error.
+
+    Example of error message without ``doc``::
+
+        everett.InvalidValueError: ValueError: invalid literal for int() with base 10:
+        'bar'; namespace=None key=foo requires a value parseable by int
+
+    Example of error message with ``doc``::
+
+        everett.InvalidValueError: ValueError: invalid literal for int() with base 10:
+        'bar'; namespace=None key=foo requires a value parseable by int
+        The port you want this to listen on.
```

Comment:
I think I would add a note here to say that's the documentation of the failing option. Something like ``doc says: `` maybe, even though I don't think that sounds too great. But I'm afraid that, in some cases, that text could be very very confusing for users.